File: Examples/Examples/Chart/ChartTypes/RangeBarColumnChart.cs
using System;
using System.Collections.Generic;
using Foundation;
using UIKit;
using TelerikUI;

namespace Examples
{
	[Register("RangeBarColumnChart")]
	public class RangeBarColumnChart : XamarinExampleViewController
	{
		TKChart chart;
		NSNumber[] lowValues;
		NSNumber[] highValues;

		public override void ViewDidLoad ()
		{
			AddOption ("Range Column", rangeColumnSelected);
			AddOption ("Range Bar", rangeBarSelected);

			base.ViewDidLoad ();

			chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			lowValues = new NSNumber[] {
				new NSNumber (33), new NSNumber (29),
				new NSNumber (55), new NSNumber (21),
				new NSNumber (10), new NSNumber (39),
				new NSNumber (40), new NSNumber (11)
			};

			highValues = new NSNumber[] {
				new NSNumber (47), new NSNumber (61),
				new NSNumber (64), new NSNumber (40),
				new NSNumber (33), new NSNumber (90),
				new NSNumber (87), new NSNumber (69)
			};

			this.rangeColumnSelected ();
		}
			
		public void rangeColumnSelected() 
		{
			chart.RemoveAllData ();

			List<TKChartRangeDataPoint> list = new List<TKChartRangeDataPoint> ();
			for (int i = 0; i < 8; i++) {
				list.Add(TKChartRangeDataPoint.RangeColumnDataPoint(new NSNumber(i), lowValues[i], highValues[i]));
			}

			TKChartRangeColumnSeries series = new TKChartRangeColumnSeries (list.ToArray());
			series.Style.PaletteMode = TKChartSeriesStylePaletteMode.UseItemIndex;
			series.Selection = TKChartSeriesSelection.DataPoint;
			chart.AddSeries(series);
			chart.ReloadData ();
		}

		public void rangeBarSelected()
		{
			chart.RemoveAllData ();
			List<TKChartRangeDataPoint> list = new List<TKChartRangeDataPoint> ();
			for (int i = 0; i < 8; i++) {
				list.Add(TKChartRangeDataPoint.RangeBarDataPoint(new NSNumber(i), lowValues[i], highValues[i]));
			}

			TKChartRangeBarSeries series = new TKChartRangeBarSeries (list.ToArray());
			series.Style.PaletteMode = TKChartSeriesStylePaletteMode.UseItemIndex;
			series.Selection = TKChartSeriesSelection.DataPoint;
			chart.AddSeries(series);
			chart.ReloadData ();
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/RangeBarColumnChart.swift
//
//  RangeBarColumnChart.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class RangeBarColumnChart: TKExamplesExampleViewController {
    let chart = TKChart()
    
    let lowValues = [33, 29, 55, 21, 10, 39, 40, 11]
    let highValues = [47, 61, 64, 40, 33, 90, 87, 69]
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        
        self.addOption("Range Column", action: rangeColumnSelected)
        self.addOption("Range Bar", action: rangeBarSelected)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        self.rangeColumnSelected()
    }
    
    func rangeColumnSelected() {
        chart.removeAllData()
        
        var items = [TKChartRangeDataPoint]()
        for i in 0..<8 {
            items.append(TKChartRangeDataPoint(x: i, low: lowValues[i], high: highValues[i]))
        }
        
        let series = TKChartRangeColumnSeries(items:items)
        series.style.paletteMode = TKChartSeriesStylePaletteMode.UseItemIndex
        series.selection = TKChartSeriesSelection.DataPoint
        chart.addSeries(series)
        chart.reloadData()
    }
    
    func rangeBarSelected() {
        chart.removeAllData()
        
        var items = [TKChartRangeDataPoint]()
        for i in 0..<8 {
            items.append(TKChartRangeDataPoint(y: i, low: lowValues[i], high: highValues[i]))
        }
        
        let series = TKChartRangeBarSeries(items:items)
        series.style.paletteMode = TKChartSeriesStylePaletteMode.UseItemIndex
        series.selection = TKChartSeriesSelection.DataPoint
        chart.addSeries(series)
        chart.reloadData()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
